src/components/Medicine/AdministerMedicine.tsx

```diff
@@ -1,165 +1,298 @@
-            className="w-full"
-            name="administration_notes"
-            placeholder={t("add_notes")}
-            value={notes}
-            onChange={({ value }) => setNotes(value)}
-            errorClassName="hidden"
-            disabled={isLoading}
-          />
-          <div className="flex flex-col gap-2 lg:max-w-min">
-            <CheckBoxFormField
-              label="Administer for a time in the past"
-              labelClassName="whitespace-nowrap"
-              name="is_custom_time"
-              value={isCustomTime}
-              onChange={({ value }) => {
-                setIsCustomTime(value);
-                if (!value) {
-                  setCustomTime(dayjs().format("YYYY-MM-DDTHH:mm"));
-                }
-              }}
-              errorClassName="hidden"
-            />
-            <DateFormField
-              name="administered_date"
-              value={customTime ? new Date(customTime) : new Date()}
-              onChange={({ value }) =>
-                setCustomTime(dayjs(value).format("YYYY-MM-DDTHH:mm"))
-              }
-              disabled={!isCustomTime}
-              min={new Date(prescription.created_date)}
-              max={new Date()}
-              errorClassName="hidden"
-              allowTime
-            />
-          </div>
+      }
+    });
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      time: new Date(),
+      note: "",
+      dosageValue: undefined,
+      allowCustomTime: false,
+    },
+  });
+
+  async function onSubmit(values: z.infer<typeof formSchema>) {
+    const time = values.time ?? new Date();
+    const doseValue = values.dosageValue ?? doseAndRate.dose_quantity?.value;
+    const doseUnit =
+      doseAndRate.dose_quantity?.unit ?? doseAndRate.dose_range?.low.unit;
+
+    const { res } = await request(routes.medicationAdministration.create, {
+      pathParams: { patientId },
+      body: {
+        encounter: consultationId,
+        request: prescription.id!,
+        status: "completed",
+        category: "inpatient",
+        medication: prescription.medication!,
+        authored_on: prescription.authored_on!,
+        recorded: new Date().toISOString(),
+        performer: [
+          {
+            actor: authUser.external_id,
+            function: "performer",
+          },
+        ],
+        occurrence_period_start: time.toISOString(),
+        occurrence_period_end: time.toISOString(),
+        note: values.note,
+        dosage: {
+          text: prescription.dosage_instruction[0].text,
+          site: prescription.dosage_instruction[0].site,
+          route: prescription.dosage_instruction[0].route,
+          method: prescription.dosage_instruction[0].method,
+          dose: {
+            value: doseValue!,
+            unit: doseUnit!,
+          },
+        },
+      },
+    });
+
+    if (res?.ok) {
+      onClose(true);
+      return;
+    }
```

Comment:
switch to useMutation. request is deprecated.

https://docs.ohc.network/docs/care/development/data-fetching-in-care/